php/libraries/LorisForm.class.inc

```diff
@@ -1877,10 +1877,10 @@ class LorisForm
     {
         $name = $this->_generateName($name);
         foreach ($array as $key => $value) {
-            $this->groupMapping[$value['name']] = array(
-                                                   'groupName' => $name,
-                                                   'index'     => $key,
-                                                  );
+            $this->groupMapping[$value['name'] ?? ''] = array(
```

Comment:
I don't know how "groupMapping" is used, but shouldn't this just not do anything (or maybe throw an exception) if $value['name'] is undefined?

Otherwise multiple undefined things will overwrite each other?

Reply:
in the current way addLabel() and createLabel() are implemented, 'name' is never defined. so throwing an exception is problemativ unless we make these two functions randomly generate names (which is fine since these fields are never submitted...)

Reply:
Okay, but why add it to the groupMapping at all?

Multiple add/createLabels will overwrite each other's index this way, so at least one is guaranteed to be wrong. It seems like it would be better to just wrap the groupMapping assignment in an if statement.

Reply:
I think the solution should not be an empty string but rather using the [_generateName()](https://github.com/xlecours/Loris/blob/469a072dcdf94dbbae521b7edbb5fd943338bd37/php/libraries/LorisForm.class.inc#L379) function to dynamically assign a name to the label. This is done for labels NOT in a group (see https://github.com/xlecours/Loris/blob/469a072dcdf94dbbae521b7edbb5fd943338bd37/php/libraries/LorisForm.class.inc#L124 )

@driusan the reason they do not ovewrite each other when multiple labels are used is that when you do something like `array[null] = something` multiple times, php generates the notice and then replaces it with `[0] = something0,  [1] = something1, [2] ... `

Reply:
No it doesn't.

```
php -a
Interactive shell

php > $x = array();
php > $x[null] = 'foo';
php > $x[null] = 'foo2';
php > print_r($x);
Array
(
    [] => foo2
)
```

Reply:
At any rate, generateName is called directly above this change. The problem is that the name generated doesn't have a value.

Reply:
examples: 
- only 1 label in group:
```
[groupMapping] => Array
        (
            [] => Array
                (
                    [groupName] => shipping_item_group_0
                    [index] => 3
                )

            [shipping_item_barcode_0] => Array
                (
                    [groupName] => shipping_item_group_0
                    [index] => 1
                )

            [shipping_item_timestamp_0] => Array
                (
                    [groupName] => shipping_item_group_0
                    [index] => 2
                )

            [comment_shipping_item_0] => Array
                (
                    [groupName] => shipping_item_group_0
                    [index] => 4
                )

        )
```

- multiple labels in group:
```
[shipping_item_group_0] => Array
                (
                    [type] => group
                    [name] => shipping_item_group_0
                    [elements] => Array
                        (
                            [0] => Array
                                (
                                    [label] => Complete Matrix box
                                    [type] => static
                                )

                            [1] => Array
                                (
                                    [label] => Tracking code
                                    [name] => shipping_item_barcode_0
                                    [class] => form-control input-sm
                                    [type] => text
                                )

                            [2] => Array
                                (
                                    [label] => Timestamp
                                    [name] => shipping_item_timestamp_0
                                    [class] => form-control input-sm
                                    [readonly] => 1
                                    [type] => text
                                )

                            [3] => Array
                                (
                                    [label] => CBSR Repository
                                    [type] => static
                                )

                            [4] => Array
                                (
                                    [label] => notes
                                    [name] => comment_shipping_item_0
                                    [class] => form-control input-sm
                                    [type] => text
                                )

                        )
```

Reply:
@driusan the _generateName is not used for this value tho it is used for the group name instead!!

Reply:
@driusan what is this function even doing ???

there are multiple solutions that result in the same behaviour
 - `if (isset($value['name'])) {` before this statement
 -  `$this->groupMapping[$value['name'] ?? $this->_generateName($value['name'])] = array(`
 -  `$this->groupMapping[$value['name'] ?? ' '] = array(`

they all seem to work

Reply:
I think the first is by far the best of those options.